components/family-studies/api/src/main/java/org/phenotips/studies/family/internal/PhenotipsFamily.java

```diff
@@ -259,12 +260,22 @@ public Pedigree getPedigree()
                 image = (BaseStringProperty) pedigreeObj.get(Pedigree.IMAGE);
 
                 if (StringUtils.isNotBlank(data.toText())) {
-                    return new DefaultPedigree(new JSONObject(data.toText()), image.toText());
+                    // internally pedigree may be stored in either "old internal" or "simpleJSON" format
+                    // for now simpleJSON may only be stored after a migration, and some of the
+                    // methods
+                    JSONObject pedigreeJSON = new JSONObject(data.toText());
+                    if (DefaultPedigree.isSupportedPedigreeFormat(pedigreeJSON)) {
+                        return new DefaultPedigree(pedigreeJSON, image.toText());
+                    } else if (NewFormatPedigree.isSupportedPedigreeFormat(pedigreeJSON)) {
+                        return new NewFormatPedigree(pedigreeJSON, image.toText());
+                    }
                 }
             } catch (XWikiException e) {
-                this.logger.error("Error reading data from pedigree. {}", e.getMessage());
+                this.logger.error("Error reading data from pedigree: [{}]", e);
```

Comment:
There will be the stacktrace, but not the right message. It looks like this:

```
[main] ERROR some.package.SomeClass - Error reading data from pedigree: [{}]
java.lang.Exception: x
	at some.package.SomeClass.main(SomeClass.java:49)
```

Notice how `[{}]` is printed uninterpolated.